modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts

```diff
@@ -150,6 +156,72 @@ export default class Ecdsa {
     return participants;
   }
 
+  /**
+   * Derive shares for a BIP-32 subkey.
+   * @param {PShare} The user's p-share.
+   * @param {NShare[]} The n-shares received from the other participants.
+   * @param {string} The BIP-32 path to derive.
+   * @returns {SubkeyShare} Returns the private p-share and n-shares to
+   * be distributed to participants at their corresponding index.
+   */
+  keyDerive(pShare: PShare, nShares: NShare[], path: string): SubkeyShare {
+    const yValues = [pShare, ...nShares].map((share) => bigIntFromBufferBE(Buffer.from(share.y, 'hex')));
+    const y = yValues.reduce((partial, share) => Ecdsa.curve.pointAdd(partial, share));
+    const u = bigIntFromBufferBE(Buffer.from(pShare.uu, 'hex'));
+    let contribChaincode = bigIntFromBufferBE(Buffer.from(pShare.chaincode, 'hex'));
+    const chaincodes = [
+      contribChaincode,
+      ...nShares.map(({ chaincode }) => bigIntFromBufferBE(Buffer.from(chaincode, 'hex'))),
+    ];
+    const chaincode = chaincodes.reduce((acc, chaincode) => (acc + chaincode) % base);
```

Comment:
This is only an issue when dealing with negatives numbers, but this would never be the case when dealing with chaincode addition. Neither operand will be interpreted as negative.

Chaincodes are used as the key in an HMAC operation during key derivation. It is not a curve operation.

Reply:
Sounds good